src/wallet/wallet.cpp

```diff
@@ -2480,6 +2480,17 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
+    reservekey.KeepKey();
```

Comment:
I think we can leave this as it is now. The only two points where this directly reserves and keeps a key is when starting the internal miner (over `-gen` or `setgenerate`, same script/key for all threads) or when calling generate.